src/include/schema.h

```diff
@@ -228,15 +229,25 @@ struct __wt_import_list {
                 WT_SESSION_LOCKED_TABLE));                                                    \
         WT_WITH_LOCK_WAIT(session, &S2C(session)->schema_lock, WT_SESSION_LOCKED_SCHEMA, op); \
     } while (0)
-#define WT_WITH_SCHEMA_LOCK_NOWAIT(session, ret, op)                               \
-    do {                                                                           \
-        WT_ASSERT(session,                                                         \
-          FLD_ISSET(session->lock_flags, WT_SESSION_LOCKED_SCHEMA) ||              \
-            !FLD_ISSET(session->lock_flags,                                        \
-              WT_SESSION_LOCKED_HANDLE_LIST | WT_SESSION_NO_SCHEMA_LOCK |          \
-                WT_SESSION_LOCKED_TABLE));                                         \
-        WT_WITH_LOCK_NOWAIT(                                                       \
-          session, ret, &S2C(session)->schema_lock, WT_SESSION_LOCKED_SCHEMA, op); \
+#define WT_WITH_SCHEMA_LOCK_NOWAIT(session, ret, op)                                               \
+    do {                                                                                           \
+        WT_ASSERT(session,                                                                         \
+          FLD_ISSET(session->lock_flags, WT_SESSION_LOCKED_SCHEMA) ||                              \
+            !FLD_ISSET(session->lock_flags,                                                        \
+              WT_SESSION_LOCKED_HANDLE_LIST | WT_SESSION_NO_SCHEMA_LOCK |                          \
+                WT_SESSION_LOCKED_TABLE));                                                         \
+        int __lock_ret;                                                                            \
+        WT_WITH_LOCK_NOWAIT(                                                                       \
+          session, ret, __lock_ret, &S2C(session)->schema_lock, WT_SESSION_LOCKED_SCHEMA, op);     \
+        if (__lock_ret != 0) {                                                                     \
+            if ((__lock_ret) == EBUSY)                                                             \
```

Comment:
I don't think we need to add extra () since we are defining it here, we know what the variable is.
```suggestion
            if (__lock_ret == EBUSY)                                                             \
```